lib/async.js

```diff
@@ -7,8 +7,16 @@ const handleError = (reject) => (err) => {
 
 module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
     return new Promise((resolve, reject) => {
+        const hookType = typeof hookFn;
```

Comment:
hook specific code doesn't belong in a module named `./lib/async`. Either rename to be hook specific, or move the hook bits out.

Reply:
rename in order 👌